src/common/utils.cpp

```diff
@@ -206,26 +206,28 @@ unsigned get_jit_profiling_flags() {
     return flag;
 }
 
-static setting_t<std::string> jit_profiling_jitdumpdir;
-dnnl_status_t init_jit_profiling_jitdumpdir(
-        const char *jitdumpdir, bool overwrite) {
-#ifdef __linux__
-    static std::mutex m;
-    std::lock_guard<std::mutex> g(m);
-
-    if (jit_profiling_jitdumpdir.initialized() && !overwrite)
-        return status::success;
-
-    if (!jitdumpdir) {
+static std::string &jit_profiling_jitdumpdir() {
+    // initialize on first use with default values from environmenet
+    static std::string s = []() -> std::string {
         char buf[PATH_MAX];
         if (getenv("JITDUMPDIR", buf, sizeof(buf)) > 0)
-            jit_profiling_jitdumpdir.set(buf);
+            return buf;
         else if (getenv("HOME", buf, sizeof(buf)) > 0)
-            jit_profiling_jitdumpdir.set(buf);
+            return buf;
         else
-            jit_profiling_jitdumpdir.set(".");
-    } else
-        jit_profiling_jitdumpdir.set(jitdumpdir);
+            return ".";
+    }();
+    return s;
+}
+
+dnnl_status_t init_jit_profiling_jitdumpdir(
+        const char *jitdumpdir, bool overwrite) {
+#ifdef __linux__
+    static std::mutex m;
+    if (overwrite) {
+        std::lock_guard<std::mutex> g(m);
+        jit_profiling_jitdumpdir() = jitdumpdir;
+    }
 
     return status::success;
 #else
```

Comment:
If we move `jit_profiling_jitdumpdir()` under `__linux__` macro then `UNUSED(jit_profiling_jitdumpdir)` is no longer needed.